web/src/features/projects/server/projectsRouter.ts

```diff
@@ -121,17 +132,39 @@ export const projectsRouter = createTRPCRouter({
         action: "delete",
       });
 
-      await ctx.prisma.project.delete({
+      // API keys need to be deleted from cache. Otherwise, they will still be valid.
+      await new ApiAuthService(ctx.prisma, redis).invalidateProjectApiKeys(
+        input.projectId,
+      );
+
+      await ctx.prisma.project.update({
         where: {
           id: input.projectId,
           orgId: ctx.session.orgId,
         },
+        data: {
+          deletedAt: new Date(),
+        },
```

Comment:
Differently than before, we soft-delete here. I think we should delete API keys here already or only fetch API keys in auth if the projectId is not soft-deleted. Otherwise, users are still able to ingest data into their project.

Reply:
Addressed in https://github.com/langfuse/langfuse/pull/4399/commits/b57459ae06b850f93d2936238a271dd89f55ec78. I just saw the invalidation on Redis and did not think about Postgres. Makes perfect sense!